core/pkg/server/handler.go

```diff
@@ -972,35 +966,48 @@ func (h *Handler) handleRequestJobInput(record *service.Record) {
 	h.fwdRecord(record)
 }
 
-func (h *Handler) handleSummary(record *service.Record, summary *service.SummaryRecord) {
-	if !h.settings.GetXSync().GetValue() {
-		// if sync is enabled, we don't need to do all this
-		runtime := int32(h.runTimer.Elapsed().Seconds())
+// handleSummary processes an update to the run summary.
+//
+// These records come from one of three sources:
+//   - Explicit updates made by using `run.summary[...] = ...`
+//   - Records from the transaction log when syncing
+//   - `updateRunTiming`
+func (h *Handler) handleSummary(
+	record *service.Record,
+	summary *service.SummaryRecord,
+) {
+	for _, update := range summary.Update {
+		err := h.runSummary.SetFromRecord(update)
+		if err != nil {
+			h.logger.CaptureError(
+				fmt.Errorf("handler: error processing summary: %v", err))
+		}
+	}
 
-		// update summary with runtime
-		summary.Update = append(summary.Update, &service.SummaryItem{
-			NestedKey: []string{"_wandb", "runtime"},
-			ValueJson: fmt.Sprintf("%d", runtime),
-		})
+	for _, remove := range summary.Remove {
+		h.runSummary.RemoveFromRecord(remove)
 	}
 
-	h.runSummary.ApplyChangeRecord(
-		summary,
-		func(err error) {
-			h.logger.CaptureError(
-				fmt.Errorf("error updating run summary: %v", err))
-		},
-	)
+	h.fwdRecord(record)
```

Comment:
currently, if we are in sync mode will we trying to write the record to the transaction log again? 
(i know we need to refactor the sync code, but this will probably break it further?)

Reply:
oh i see we have the skipSummary! Got it!

Reply:
wouldn't that apply to all the records written to the transaction log? for example consider `OutputRaw`---in `handleOutputRaw` we simply forward it without modifying the control, but if we read it from the transaction log, then its control was necessarily `Local`.

more on the topic of this particular code, I would say it's up to the caller to set the control

Reply:
Yeah agreed that it is up to the caller! i think when we fix the sync service, we want it to be responsible to setting the control to local to avoid writing to the transaction log